File: lib/features/saliks/domain/entities/approval_status.dart
enum ApprovalStatus {
  pending,
  approved,
  rejected;

  static ApprovalStatus fromString(String? value) {
    switch (value?.trim().toLowerCase()) {
      case 'pending':
        return ApprovalStatus.pending;
      case 'rejected':
        return ApprovalStatus.rejected;
      default:
        return ApprovalStatus.approved;
    }
  }

  String toFirestore() => name;
}

File: app/src/main/java/com/example/salik_management_system/features/saliks/domain/model/ApprovalStatus.kt
package com.example.salik_management_system.features.saliks.domain.model

enum class ApprovalStatus {
    Pending,
    Approved,
    Rejected;

    fun toFirestore(): String = when (this) {
        Pending -> "pending"
        Approved -> "approved"
        Rejected -> "rejected"
    }

    companion object {
        fun fromString(value: String?): ApprovalStatus {
            return when (value?.trim()?.lowercase()) {
                "pending" -> Pending
                "rejected" -> Rejected
                else -> Approved
            }
        }
    }
}
